File: Code Challenges/Week 5/Pop, Shift, & Unshift Elements.js
/*

Write a function that takes an array of strings as input and returns a new array where the first element is moved to the end of the array. You must use both shift and push methods to achieve this.

*/

function shiftFirstToEnd(arr) {
    let firstElement = arr.shift();
    arr.push(firstElement);
    return arr;
   }
   
   console.log(shiftFirstToEnd(["apple", "banana", "orange"])); // ["banana", "orange", "apple"]
   
   
   /*
   
   Write a function that takes an array of numbers as input and returns a new array where the last element is moved to the beginning of the array. You must use both pop and unshift methods to achieve this.
   
   */
   
   function popLastToStart(arr) {
    let lastElement = arr.pop();
    arr.unshift(lastElement);
    return arr;
   }
   
   console.log(popLastToStart([1, 2, 3, 4, 5])); // [5, 1, 2, 3, 4]

File: Code Challenges/Week 5/Pop, Shift, & Unshift Elements.py
# Write a function that takes a list of strings as input and returns a new list where the first element is moved to the end of the list. You must use both shift and push methods to achieve this. Additionally, handle edge cases such as empty lists or lists with fewer than two elements.

def move_first_to_end(lst):
    if len(lst) < 2:
        return lst
    
    # Create a new list to store the result
    result = []
    
    # Push all elements except the first one to the result list
    for i in range(1, len(lst)):
        result.append(lst[i])
    
    # Push the first element to the end of the result list
    result.append(lst[0])
    
    return result

# Test cases
print(move_first_to_end(['a', 'b', 'c']))  # ['b', 'c', 'a']
print(move_first_to_end(['x']))           # ['x']
print(move_first_to_end([]))              # []